crates/forge/tests/it/cheats.rs

```diff
@@ -7,14 +7,16 @@ use crate::{
         TEST_DATA_MULTI_VERSION,
     },
 };
+use alloy_primitives::U256;
 use foundry_config::{fs_permissions::PathPermission, FsPermissions};
 use foundry_test_utils::Filter;
 
-/// Executes all cheat code tests but not fork cheat codes or tests that require isolation mode
+/// Executes all cheat code tests but not fork cheat codes or tests that require isolation mode or
+/// specific seed.
 async fn test_cheats_local(test_data: &ForgeTestData) {
     let mut filter = Filter::new(".*", ".*", &format!(".*cheats{RE_PATH_SEPARATOR}*"))
         .exclude_paths("Fork")
-        .exclude_contracts("Isolated");
+        .exclude_contracts("(Isolated|WithSeed)");
```

Comment:
please add it to `if cfg!(feature = "isolate-by-default")` branch below too, tests with `--features=isolate-by-default` are failing because we're overriding it there

Reply:
sorry, missed running the test with feature set locally, added here https://github.com/foundry-rs/foundry/pull/8807/commits/02086f94db6b6db262ded77b53d5c480f9f4fc8a (had to add mock function test too)

Reply:
> had to add mock function test too

hmm, wondering why would it break with `--isolate`?

Reply:
ah, I see, it's because we're ignoring `bytecode_address` when handling isolated calls https://github.com/foundry-rs/foundry/blob/2c73013a01264e8577e728bee33961845d126963/crates/evm/evm/src/inspectors/stack.rs#L745

Reply:
not sure how to handle this now, probably not a big deal

could be solved by instead of altering `bytecode_address` changing target address bytecode for duration of the call, though it would introduce new edge cases

Reply:
got it, thanks for clarifying! do you think we should be looking into supporting it as well or good to have it only for non isolating calls for now?